src/category/category.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, ConflictException, NotFoundException } from "@nestjs/common";
+import { CategoryRepository } from "./category.respository";
+import { CreateCategoryDTO } from "./dto/create-category.dto";
+import { User } from "src/user/user.model";
+import { Category } from "./category.model";
+import { MessageUtil } from "src/shared/util/message.util";
+
+@Injectable()
+export class CategoryService {
+
+    constructor(
+        private readonly categoryRepository: CategoryRepository
+    ) { }
+
+    async createCategory(
+        currentUser: User,
+        dto: CreateCategoryDTO) {
+        const category = new Category();
+
+        if (await this.categoryRepository.findByName(dto.name)) {
+            throw new ConflictException(MessageUtil.EXIST_CATEGORY_NAME);
```

Comment:
Hi Emmanuel, please follow the convention for the MessageUtil message when a resource already exists. It should be <RESOURCE>_ALREADY_ EXISTS.
So in this case CATEGORY_ALREADY_EXISTS